yt/tests/test_load_errors.py

```diff
@@ -73,23 +74,87 @@ def test_load_simulation_unidentified_data_file(tmp_path_with_empty_file):
         )
 
 
-def test_load_ambiguous_data(tmp_path):
-    # we deliberately setup a situation where two Dataset subclasses
-    # that aren't parents are consisdered valid
-    class FakeDataset(Dataset):
+@pytest.fixture()
+def ambiguous_dataset_classes():
+    # We deliberately setup a situation where two Dataset subclasses
+    # that aren't parents are consisdered valid.
+    # We implement the bare minimum for these classes to be actually
+    # loadable in order to test hints.
+    class MockHierarchy(GridIndex):
```

Comment:
I think we may want to consider lifting these in the start of an effort to use them elsewhere.

Reply:
sorry I'm not following, what do you mean ?

Reply:
On re-reading, my original idea of making this not specific to ambiguous dataset classes and instead have general-purpose mocks is not necessary.